Makefile

jandrovins opened the pull request:
**Regarding Makefile changes**: changed some variables according to some standards [https://www.gnu.org/prep/standards/html_node/Makefile-Conventions.html](url). Variables ending with "dir" (e.g. "SRCdir") do not follow a standard other than being easily recognizable and self-explaining. I think it would be OK if they did not have the "dir" part.

**Regarding folder structure changes**:  even though there is no standard for a C project folder structure (other than being intuitive ) I believe the changes are for good. Old structure is very complicated for our project. 

There is no need for a folder named "c" under "src" for placing C source code. Isn't it the purpose of "src"? After checking some other C projects this change seemed pretty OK. Same arguments for test/c.

Also, it is more convenient to have an "include" directory in our top directory. Header files are easier to find. 

Both changes deacreses the amount of subfolders from our folder structure, and makes it simpler. You will easily recognize it by running "tree" on our top directory :)

BalmungSan commented on src/main.c:
```diff
@@ -1,6 +1,6 @@
-#include "data.h"
-#include "functions.h"
+#include <data.h>
+#include <functions.h>
```
The standard is to only use `<>` for standard or global includes and ``""` for user defined includes.
https://stackoverflow.com/questions/21593/what-is-the-difference-between-include-filename-and-include-filename

jandrovins commented:
- >     * I do not see any necessity of having `ALL_CFLAGS`, specially since `CFLAGS` is empty.
 The reason for this is that the user should be able to change the value of `CFLAGS` from the CLI and still    compile successfully. So under `ALL_CFLAGS` are the flags that are strictly needed for our software, but also can be overwritten by `CFLAGS` if the user wants to, by being at the end of `ALL_CFLAGS`.


- >     * The best standard is to follow the same standard on all places. `program\_name` should be renamed as `PROGRAM\_NAME`. Also, and more importantly, all the **dirs** are wrong, if going _SCREAMING_SNAKE_CASE_ for all other things, then ir makes sense to follow it there too.
I changed `program_name` and removed the "dir" part from the variables. Is that what you meant?

- >     * You added a binary _(the tests executable)_ to the repo, and moved the generated files outside the ignored **bin** folder. That is a real bad practice.
Fixed.
- >     * The equals on the variables seems like you wanted to align them _(which I like)_ but they do not. Can not check right now, but I am sure that you mixed white spaces with tabs.
Fixed.

- >     * Finally, now it seems odd to have `src/main` & `src/test` what about just removing `src` and leaving a top level directories `main`, `include` & `test`?
I think the structure I proposed is more intuitive. All source code should be under `src`. Tests depend on `functions.c`, so for me it makes sense that `test` is under `src` and having a directory just for main would make no sense. The only reason for `include` to be on our top directory would be that the structure would be simpler. I think is unnecessary for `include` to be under source. 

- Also I think `bin` should be reserved only to binaries, so what do you think on moving `obj` directory to our top directory?

BalmungSan commented:
> The reason for this is that the user should be able to change the value of CFLAGS from the CLI and still compile successfully. So under ALL_CFLAGS are the flags that are strictly needed for our software, but also can be overwritten by CFLAGS if the user wants to, by being at the end of ALL_CFLAGS.

Fair enough.

> I changed program_name and removed the "dir" part from the variables. Is that what you meant?

I am not sue if a like `MAIN` the idea was that we should rename it from `main` to `2DPartInt` or something like that. Thus `PROGRAM_NAME` or `EXE_NAME` or something like that, sound more intuitive IMHO.
Also, if you want, you can rename the dirs variables to `SRC_DIR` or `SRC_FOLDER` which would make them more readable.

> I think the structure I proposed is more intuitive. All source code should be under src. Tests depend on functions.c, so for me it makes sense that test is under src and having a directory just for main would make no sense. The only reason for include to be on our top directory would be that the structure would be simpler. I think is unnecessary for include to be under source.

Again, for me, the includes are also source. Thus, why moving it outside of `src`?
Also, I mean something like

.
  - include/
  - main/
  - test/
  - Makefile

All on the top level.

> Also I think bin should be reserved only to binaries, so what do you think on moving obj directory to our top directory?

Well, it could be, we would just need to change the _gitignore_ file, it only worry me that having another top level file for intermediate values may clutter too much the working space.
For example, in **Java** all intermediate files _(like .class)_ are stored in a sub-folder of `target`, which is the folder where the generate Jar is stored too.

jandrovins commented:
Updated with most of your suggestions. I would make only one change to what you proposed:

> Also, I mean something like
> 
> .
> 
>     * include/
> 
>     * main/
> 
>     * test/
> 
>     * Makefile
I like the structure you suggested for being simple and clean. I would change only one thing. I suggest `src` instead of `main`, due to the fact that is a more intuitive name for a place to store core source files, and is what most projects use.

BalmungSan commented:
> I like the structure you suggested for being simple and clean. I would change only one thing. I suggest src instead of main, due to the fact that is a more intuitive name for a place to store core source files, and is what most projects use.

Yes, but most projects have also a `main` folder to separate **main** _sources_ from **tests** _sources_. That is why I proposed the original structure.

jandrovins commented:
I made the changes we talked about this morning, pretty sure I'm not forgotting something. Tell me otherwise.

Regarding cpp folder: since one cannot push an empty folder I would have to create a file with a dot inside in order to push it, so I thought it would be better that we wait until a C++ source file is pushed.

BalmungSan commented on .gitignore:
```diff
@@ -5,3 +5,4 @@
 *.exe
 *.out
 /bin/
+/obj/
```
It is `build` instead of `/obj/` but I will change that on #12
